pkg/controllers/flannelmigration/network_migrator.go

```diff
@@ -0,0 +1,216 @@
+		// Canal creates tunnel device but with no bridge. It uses Calico CNI.
+		cmd = fmt.Sprintf("ip link show flannel.%d || exit 0 && { echo '%s' > /host/%s/%s ; ip link delete flannel.%d; } && exit 0 || exit 1",
+			m.config.FlannelVNI, dummyCNI, m.config.CNIConfigDir, m.calicoCNIConfigName, m.config.FlannelVNI)
+	} else {
+		// Flannel creates cni0 bridge and tunnel device. It delegates to bridge CNI.
+		cmd = fmt.Sprintf("{ ip link show cni0; ip link show flannel.%d; } || exit 0 && { echo '%s' > /host/%s/%s ; ip link delete cni0 type bridge; ip link delete flannel.%d; } && exit 0 || exit 1",
+			m.config.FlannelVNI, dummyCNI, m.config.CNIConfigDir, m.calicoCNIConfigName, m.config.FlannelVNI)
+	}
+
+	// Run a remove-flannel pod with specified nodeName, this will bypass kube-scheduler.
+	// https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#nodename
+	pod := k8spod("remove-flannel")
+	podLog, err := pod.RunPodOnNodeTillComplete(m.k8sClientset, namespaceKubeSystem, m.calicoImage, node.Name, cmd, m.config.CNIConfigDir, true, true)
+	if podLog != "" {
+		log.Infof("remove-flannel pod logs: %s.", podLog)
+	}
+
+	// Wait twice as long as default sync period so kubelet has picked up dummy CNI config.
+	// We probably need something better here but currently no API available for us to know
+	// if kubelet sees new config.
+	time.Sleep(2 * defaultSyncConfigPeriod)
+	return err
+}
+
+// Drain node, remove Flannel and setup Calico network for a node.
+func (m *networkMigrator) setupCalicoNetworkForNode(node *v1.Node) error {
+	log.Infof("Setting node label to disable Flannel daemonset pod on %s.", node.Name)
+	// Set node label so no Flannel pod can be scheduled on this node.
+	// Flannel pod currently running on the node starts to be evicted as the side effect.
+	n := k8snode(node.Name)
+	err := n.addNodeLabels(m.k8sClientset, nodeNetworkNone)
+	if err != nil {
+		log.WithError(err).Errorf("Error adding node label to disable Flannel network for node %s.", node.Name)
+		return err
+	}
+
+	// Cordon and Drain node. Make sure no pod (except daemonset pod or pod with nodeName selector) can run on this node.
+	err = n.Drain()
+	if err != nil {
+		log.WithError(err).Errorf("failed to drain node %s", node.Name)
+		return err
+	}
+
+	log.Infof("Removing flannel tunnel device/routes on %s.", node.Name)
+	// Remove Flannel network from node.
+	// Note Flannel vxlan tunnel device (flannel.1) is created by Flannel daemonset pod (not Flannel CNI)
+	// Therefor if Flannel daemonset pod can not run on this node, the tunnel device will not be recreated
+	// after we delete it.
+	err = m.removeFlannelNetworkAndInstallDummyCalicoCNI(node)
+	if err != nil {
+		log.WithError(err).Errorf("failed to remove flannel network on node %s", node.Name)
+		return err
+	}
+
+	log.Infof("Deleting non-host-networked pods on %s.", node.Name)
+	// Delete all pods on the node which is not host networked.
+	err = n.deletePodsForNode(m.k8sClientset, func(pod *v1.Pod) bool {
```

Comment:
I checked `kubectl drain` code, it has some logic that tries to `evict` pods instead of `delete`. It also support some Flags (e.g. grace-period) which may be useful for user later.